pkg/bpf/map_linux_test.go

```diff
@@ -774,3 +776,39 @@ func (s *BPFPrivilegedTestSuite) TestCreateUnpinned(c *C) {
 	c.Assert(err, IsNil)
 	c.Assert(got, checker.DeepEquals, v)
 }
+
+func BenchmarkMapLookup(b *testing.B) {
```

Comment:
Q: What's the context for this benchmark, are we proving that the ebpf-go has better overhead over the old Cilium bpf.Map?

Reply:
Yes, there is a rather large performance regression due to getting rid of unsafe.Pointer in favor of reflect-based unmarshaling in most of the maps. We're currently addressing this in ebpf-go by adding a binary.Writer pool on the key marshaling path in the short term and investigating in which cases we can pass unsafe.Pointers to the kernel internally. The latter will require a few more cycles and won't make it in time for the 1.14 freeze.

Reply:
Upstream PR merged and pulled into vendor in this PR.